### PR TITLE
DAOS-8458 container: remove uuid handle from cont iv code

### PR DESCRIPTION
**Pull request by mchaarawi**
remove the uuid handle from the oid allocator code for iv. This makes allocation more efficient when using the
same container but different handles for that container and also avoid confusion that the allocator is per container and not per oid.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

**Review by liw — APPROVED** [review]

File: src/container/srv_target.c [hunk 1]
```diff
@@ -2201,9 +2201,8 @@ cont_oid_alloc(struct ds_pool_hdl *pool_hdl, crt_rpc_t *rpc)
 	sgl.sg_nr_out = 0;
 	sgl.sg_iovs = &iov;
 
-	rc = oid_iv_reserve(pool_hdl->sph_pool->sp_iv_ns,
-			    in->coai_op.ci_pool_hdl, in->coai_op.ci_uuid,
-			    in->coai_op.ci_hdl, in->num_oids, &sgl);
+	rc = oid_iv_reserve(pool_hdl->sph_pool->sp_iv_ns, in->coai_op.ci_pool_hdl,
+			    in->coai_op.ci_uuid, in->num_oids, &sgl);
```
Comment (wangdi1) [inline src/container/srv_target.c:2205]: probably use pool uuid, instead of pool handle uuid as well?

**Review by wangdi1 — DISMISSED** [review]

File: src/container/srv_target.c (hunk 1 above)
Comment (mchaarawi) [reply src/container/srv_target.c:2205]: yea makes sense, let me push an update

**Review by wangdi1 — APPROVED** [review]
Thanks!

**Review by liw — APPROVED** [review]

**Comment by mchaarawi** [comment]
had to make a small change so CI does not complain on the cmocka test name :-/

**Review by wangdi1 — APPROVED** [review]

**Review by liw — APPROVED** [review]

**Review by liw — APPROVED** [review]

**Pull request merged**